### ISSUE TITLE
`TypeGuard[...]` is a subtype of  `bool` but not `int` since 1.1.374

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#8769]
**Describe the bug**

Any value with type of `TypeGuard[...]` can be assigned to a variable with type of `bool` but not a variable with type of `int`, and `int` methods are not available.

The same goes for `TypeIs[...]`.

**Code or Screenshots**

```python
from typing import TypeGuard

def is_str(_: object) -> TypeGuard[str]: ...

v_0 = is_str(object())
v_1: bool = v_0  # Passes.
v_2: int = v_0  # Used to pass, but is failing.
v_3 = v_0 & v_0  # Used to pass, but is failing.
```

```python
import numpy.typing
import pandas

def foo(
    a: numpy.typing.NDArray[numpy.float64] | float,
    b: numpy.typing.NDArray[numpy.float64] | float,
) -> numpy.typing.NDArray[numpy.bool_] | bool:
    return pandas.notna(a) & pandas.notna(b)  # Used to pass, but is failing.
```

**VS Code extension or command-line**

Pyright 1.1.374 and later.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release. I've also updated the unit tests to cover these cases, so this regression shouldn't go undetected again.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.377.
